packages/flutter/lib/src/painting/decoration_image.dart

```diff
@@ -424,6 +424,10 @@ void debugFlushLastFrameImageSizeInfo() {
 ///
 ///  * `image`: The image to paint onto the canvas.
 ///
+///  * `devicePixelRatio`: The device pixel ratio of the [FlutterView] into
+///     which the image will be drawn. Can be obtained via
+///     `View.of(context).devicePixelRation`.
```

Comment:
note to self: mention `MediaQuery.devicePixelRatioOf(context)` as well

Reply:
```suggestion
///     `View.of(context).devicePixelRatio`.
```

Reply:
Can this get defaulted somehow?

For example, just default it to the first view in the collection of views?

Reply:
If we default this, the assert check in `paintImage` becomes more or less meaningless, no? It needs to know the exact dpr in order to figure out whether the current image dimensions are overkill or not.

Reply:
We could just use the max device pixel ratio of all views as an upper bound, though. That will greatly simplify things.